### PR TITLE
feat(scaffold): add typescript support

### PR DESCRIPTION
**Pull request by dhruvdutt**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

File: packages/webpack-scaffold/index.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+export function createArrowFunction(value: Function): string {
+	return `() => '${value}'`;
+}
+
+export function createRegularFunction(value: Function): string {
+	return `function () {\n return '${value}'\n}`;
+}
+
+export function createDynamicPromise(arr: Function[]): string {
```
Comment (ev1stensberg) [inline packages/webpack-scaffold/index.ts]: not always an array, could be arr or string

File: packages/webpack-scaffold/index.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	type: string;
+	name: string;
+	message: string;
+}
+
+interface IList extends IBaseObj {
+	choices?: string[];
+}
+
+interface IInput extends IBaseObj {
```
Comment (ev1stensberg) [inline packages/webpack-scaffold/index.ts]: better convention here. `Inq_Input` etc maybe

File: packages/webpack-scaffold/index.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import jscodeshift from "jscodeshift";
+
+interface IBaseObj {
```
Comment (ematipico) [inline packages/webpack-scaffold/index.ts]: I think we should change the name. `IBaseObj` sounds too generic

File: packages/webpack-scaffold/index.ts (hunk 2 above)
Comment (ematipico) [reply packages/webpack-scaffold/index.ts]: What "Inq" stands for?

Comment (dhruvdutt) [reply packages/webpack-scaffold/index.ts]: I think Inquirer?

File: packages/webpack-scaffold/index.ts (hunk 3 above)
Comment (dhruvdutt) [reply packages/webpack-scaffold/index.ts]: This interface would be scoped to this file so I thought to name it that way. What do you suggest?

Comment (ematipico) [reply packages/webpack-scaffold/index.ts]: `IScaffoldBase` or `IScaffoldBaseObject`? Something like this. What do you think?

**Pull request merged**